plugin.video.vimeo/resources/lib/kodi/vfs.py

```diff
@@ -21,7 +21,7 @@ def read(self, filename):
 
     def write(self, filename, string):
         filepath = os.path.join(self.path, filename)
-        file = xbmcvfs.File(filepath, 'w')
+        file = xbmcvfs.File(filepath, "w")
```

Comment:
why not use `with xbmcvfs.File(filepath, "w") as file:` then you don't need the `file.close()` and it is safer because it automatically closes the file handle.

Reply:
I'm not sure this would work, since the `with` statement requires an `__exit__` method on the object (which is `xbmcvfs.File` in this case): https://docs.python.org/3/reference/compound_stmts.html#the-with-statement

 If I'm changing the code, I'm getting the following exception:

```
2020-03-31 18:32:35.442 T:140412491732736   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.AttributeError'>
                                            Error Contents: __exit__
                                            Traceback (most recent call last):
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/addon.py", line 3, in <module>
                                                plugin.run()
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/resources/plugin.py", line 66, in run
                                                collection = listItems.from_collection(api.channel(channel_id))
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/resources/lib/vimeo/api.py", line 109, in channel
                                                res = self._do_api_request("/channels/{id}/videos".format(id=channel_id), params)
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/resources/lib/vimeo/api.py", line 136, in _do_api_request
                                                return self.api_client.get(path, params=params).json()
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/resources/lib/vimeo/api.py", line 61, in api_client
                                                self.api_access_token_cache_duration
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/resources/lib/kodi/cache.py", line 15, in get
                                                file = self.vfs.read(filename)
                                              File "/home/osboxes/.kodi/addons/plugin.video.vimeo/resources/lib/kodi/vfs.py", line 15, in read
                                                with xbmcvfs.File(filepath, "w") as file:
                                            AttributeError: __exit__
                                            -->End of Python script error report<--
```

Am I doing something wrong? This is the code:

```python
    def read(self, filename):
        filepath = os.path.join(self.path, filename)
        if xbmcvfs.exists(filepath):
            with xbmcvfs.File(filepath, "w") as file:
                string = file.read()
            return string
        else:
            return None
```

Reply:
Ok, this can only be done in v19. :slightly_smiling_face: 

https://github.com/xbmc/xbmc/commit/0775d520a166e07322ffdd092d5c8e2758b1ddc2

I'll remember to change this when I submit the addon to the Matrix branch.